test/object-store/sync/app.cpp

```diff
@@ -2970,6 +2982,17 @@ TEST_CASE("app: sync integration", "[sync][pbs][app][baas]") {
             REQUIRE(!user->is_logged_in());
         }
 
+        SECTION("User is left logged out if logged out while the refresh is in progress") {
+            REQUIRE(user->is_logged_in());
+            transport->request_hook = [&](const Request&) {
+                user->log_out();
+            };
+            SyncTestFile config(app, partition, schema);
+            auto r = Realm::get_shared_realm(config);
+            REQUIRE_FALSE(user->is_logged_in());
+            REQUIRE(user->state() == SyncUser::State::LoggedOut);
```

Comment:
should this test wait for the access token response before checking the user state?

Reply:
oh never mind, we use a synchronous transport